### PR TITLE
Merge in CF 1.8 implementation

### PR DESCRIPTION
**Pull request by benjwadams**
CF 1.8 implementation.   Needs to be merged, along with any desired changes, prior to next release.

### CODE REVIEW

**Review by greenteadigital — COMMENTED** [review]
cf.py:4102
"boundary_variable" is not defined

ioos.py:959
"good_result" is not defined

**Comment by benjwadams** [comment]
> cf.py:4102
   "boundary_variable" is not defined

These were introduced in other commits unrelated to CF 1.8.  I am very much in favor of fixing these and adding the proper test coverage where needed prior to the next tagged release, even possibly revisiting #641.  However, I think they should be dealt with in a separate issue, as the commit change set for this PR didn't add them.

In terms of splitting out the CF versions into separate modules, I'm in favor of it overall as long as we still import the classes to the `cf.py` module.  There are a number of plugins for compliance checker which rely on the classes being importable from that module.

**Comment by greenteadigital** [comment]
OK, I can live w/ that.

File: .github/workflows/default-tests.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
     - name: Python ${{ matrix.python-version }}
       shell: bash -l {0}
       run: |
-        conda create --name TEST python=${{ matrix.python-version }} pip --file requirements.txt --file test_requirements.txt --strict-channel-priority
+        conda create --name TEST python=${{ matrix.python-version }} pip "libnetcdf<4.8.0" --file requirements.txt --file test_requirements.txt --strict-channel-priority
```
Comment (ocefpaf) [inline .github/workflows/default-tests.yml:29]: We should probably revert this change. We need to unpin libnetcdf.

Comment (benjwadams) [reply .github/workflows/default-tests.yml:29]: Sorry, I accidentally merged in some stuff related to 1.9 into develop, which I didn't mean to do.  Some of the commits above are my attempts to rectify some of the issues.  Others are just from merging master.  I'll unpin this.

**Comment by benjwadams** [comment]
@greenteadigital , will break apart the CF modules once I'm satisfied tests are working again.  Some of the integration tests seem to have had URLs change and now cause failures.

**Comment by benjwadams** [comment]
Looks like something is breaking in `ncgen` for generating string variables.

**Comment by benjwadams** [comment]
OK @greenteadigital, I have refactored the versioned CF checker classes into their own respective modules, which are then imported into `cf.py` for the sake of backwards compatibility.  Looking at this, I think it may make sense to import them into `__init__.py` instead, but that would break backwards compatibility with some plugins.  It'd be "nicer" to be able to do `from compliance_checker.cf import CF1_8Check` instead of `from compliance_checker.cf.cf import CF1_8Check`.  Perhaps if we do decide to move to `__init__.py`, we can have a deprecation warning for the import from `cf.py`.
